### DISCUSSION TITLE
[@types/reach__router] type for location.state in useLocation

### DISCUSSION THREAD

**Discussion by cavalier38** [DefinitelyTyped/DefinitelyTyped#58328] [Issues with a @types package] [answered]
Can the type definition of useLocation be changed that is accepts an optional state type?
See also reach/router#414
e.g.
`export function useLocation<S = unknown>(): WindowLocation<S>; `

[a comment by an automated account is not shown]

**Comment by Eyas** [comment]
Sounds about right! Feel free to send a contribution to fix this

**Comment by cavalier38** [comment, marked as answer]
running npm test gives me the following error:

ERROR: 175:60  no-unnecessary-generics  Type parameter S is used only once. See: https://github.com/Microsoft/dtslint/blob/master/docs/no-unnecessary-generics.md

The link says that the generic is not good, as it is only used once.
So, instead of using:
`const location = useLocation<MyState>();`
It suggests to use:
`const location = useLocation() as WindowLocation<MyState>;`

If that is correct, there is no change to the types defined, only a documented answer should be enough.

Please advice.

**Reply by martin-badin** [reply]
you don't need to use generics if you are using the type once. The suggestion is correct in this case.
`const location = useLocation() as WindowLocation<MyState>;`
